trackersite/tracker/views.py

```diff
@@ -233,21 +283,50 @@ def create_ticket(request):
             if ticket.topic.ticket_expenses:
                 expeditures.instance = ticket
                 expeditures.save()
-            
+            if ticket.topic.ticket_preexpenses:
+                preexpeditures.instance = ticket
+                preexpeditures.save()
+
             messages.success(request, _('Ticket %s created.') % ticket)
             return HttpResponseRedirect(ticket.get_absolute_url())
     else:
         initial = {'event_date': datetime.date.today()}
         if 'topic' in request.GET:
             initial['topic'] = request.GET['topic']
+        if 'ticket' in request.GET:
+            ticket = get_object_or_404(Ticket, id=request.GET['ticket'])
+            initial['summary'] = ticket.summary
+            initial['topic'] = ticket.topic
+            initial['description'] = ticket.description
+            initial['deposit'] = ticket.deposit
         ticketform = TicketForm(initial=initial)
         mediainfo = MediaInfoFormSet(prefix='mediainfo')
-        expeditures = ExpeditureFormSet(prefix='expediture')
-    
+        initialExpeditures = []
+        if 'ticket' in request.GET:
+                for e in Expediture.objects.filter(ticket=ticket):
+                        initialE = {}
+                        initialE['description'] = e.description
+                        initialE['amount'] = e.amount
+                        initialE['wage'] = e.wage
+                        initialExpeditures.append(initialE)
+        ExpeditureFormSet = preexpeditureformset_factory(extra=2+len(initialExpeditures), can_delete=False)
+        expeditures = ExpeditureFormSet(prefix='expediture', initial=initialExpeditures)
+        initialPreexpeditures = []
+        if 'ticket' in request.GET:
+            for pe in Preexpediture.objects.filter(ticket=ticket):
+                initialPe = {}
+                initialPe['description'] = pe.description
+                initialPe['amount'] = pe.amount
+                initialPe['wage'] = pe.wage
+                initialPreexpeditures.append(initialPe)
+        PreexpeditureFormSet = preexpeditureformset_factory(extra=2+len(initialPreexpeditures), can_delete=False)
+        preexpeditures = PreexpeditureFormSet(prefix='preexpediture', initial=initialPreexpeditures)
```

Comment:
Nejde tuhle copy&paste ságu nějak zjednodušit?

Reply:
Murphyho zákon o programování: 
Každý program obsahuje jeden chybný řádek. Každý program jde zkrátit o jeden řádek. Z toho plyne, že každý program jde zkrátit na jeden řádek, který je chybný.

Z toho plyne, že nepochybně to nějak zjednodušit a zkrátit půjde. Otázkou je, zda to nebude zkrácení na úkor srozumitelnosti.